### DISCUSSION TITLE
Allow.authenticated

### DISCUSSION THREAD

**Discussion by danielchernilov** [noam-honig/radweb#373] [Q&A]
Hi,
How can one control the state of Allow.authenticated as demonstrated in the following [video](https://youtu.be/9lWQwAUcKEM?si=Z-Ej-V6MclVMAI9L) at 6:31? Are there specific methods within the Remult library that enable users to manage the authentication status using Allow.authenticated?

**Comment by noam-honig** [comment]
Checkout the auth tutorial step:
https://remult.dev/tutorials/react/auth.html

On the backend, you can tell remult how to figure out the current user based on the request:
```ts
// src/server/api.ts

//...

export const api = remultExpress({
  //...
  getUser: req => req.session!["user"]
})
```

In the frontend, once you identify the user, simply set the `user` property of remult to them:
```ts
  const signIn = async (e: FormEvent) => {
    e.preventDefault()
    const result = await fetch("/api/signIn", {
      method: "POST",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify({ username })
    })
    if (result.ok) {
      remult.user = await result.json()  // --> set the user in the frontend
      setSignedIn(true)
      setUsername("")
    } else {
      alert(await result.json())
    }
  }
```

This will also set the roles that'll be used for allow methods and in the call to `remult.isAllowed`
